src/modules/LimitOrders.ts

```diff
@@ -0,0 +1,353 @@
+      { t: 'bytes32', v: stringToBytes32(EIP712_DOMAIN_STRING) },
+      { t: 'bytes32', v: stringToBytes32('LimitOrders') },
+      { t: 'bytes32', v: stringToBytes32('1.0') },
+      { t: 'uint256', v: chainId },
+      { t: 'bytes32', v: addressToBytes32(contractAddress) },
+    );
+  }
+
+  public orderToBytes(
+    order: LimitOrder,
+    includeSignature: boolean = true,
+  ): string {
+    let byteArray = []
+      .concat(argToBytes(order.makerMarket))
+      .concat(argToBytes(order.takerMarket))
+      .concat(argToBytes(order.makerAmount))
+      .concat(argToBytes(order.takerAmount))
+      .concat(argToBytes(order.makerAccountOwner))
+      .concat(argToBytes(order.makerAccountNumber))
+      .concat(argToBytes(order.takerAccountOwner))
+      .concat(argToBytes(order.takerAccountNumber))
+      .concat(argToBytes(order.expiration))
+      .concat(argToBytes(order.salt));
+    if (includeSignature && order.signature) {
+      byteArray = byteArray.concat(hexToBytes(order.signature));
+    }
+    return bytesToHex(byteArray);
+  }
+
+  // ============ Private Helper Functions ============
+
+  private orderHashToCancelOrderHash(
+    orderHash: string,
+  ): string {
+    return soliditySha3(
+      { t: 'string', v: 'cancel' },
+      { t: 'bytes32', v: orderHash },
+    );
+  }
+
+  private async ethSignTypedOrderInternal(
+    order: LimitOrder,
+    rpcMethod: string,
+    hashOptions: any = {},
+  ): Promise<string> {
+    const contractAddress =
+      hashOptions.contractAddress || this.contracts.limitOrders.options.address;
+    const chainId = hashOptions.chainId || this.networkId;
+    const domainData = {
+      chainId,
+      name: 'LimitOrders',
+      version: '1.0',
+      verifyingContract: contractAddress,
+    };
+    const orderData = {
+      makerMarket: order.makerMarket.toFixed(0),
+      takerMarket: order.takerMarket.toFixed(0),
+      makerAmount: order.makerAmount.toFixed(0),
+      takerAmount: order.takerAmount.toFixed(0),
+      makerAccountOwner: order.makerAccountOwner,
+      makerAccountNumber: order.makerAccountNumber.toFixed(0),
+      takerAccountOwner: order.takerAccountOwner,
+      takerAccountNumber: order.takerAccountNumber.toFixed(0),
+      expiration: order.expiration.toFixed(0),
+      salt: order.salt.toFixed(0),
+    };
+    const data = {
+      types: {
+        EIP712Domain: EIP712_DOMAIN_STRUCT,
+        LimitOrder: EIP712_ORDER_STRUCT,
+      },
+      domain: domainData,
+      primaryType: 'LimitOrder',
+      message: orderData,
+    };
+    const sendAsync = promisify(this.web3.currentProvider.send).bind(this.web3.currentProvider);
+    const response = await sendAsync({
+      method: rpcMethod,
+      params: [order.makerAccountOwner, data],
+      jsonrpc: '2.0',
+      id: new Date().getTime(),
+    });
+    if (response.error) {
+      throw new Error(response.error.message);
+    }
+    return `0x${stripHexPrefix(response.result)}0${SIGNATURE_TYPES.NO_PREPEND}`;
```

Comment:
don't we want to include the prefix? otherwise metamask like freaks out and shows a huge warning

Reply:
It just strips it if it exists and then explicitly inserts it